context/src/main/java/eu/maveniverse/maven/mima/context/ContextOverrides.java

```diff
@@ -389,6 +475,20 @@ public static Builder create() {
             return new Builder();
         }
 
+        /**
+         * Sets basedir path, it must be an existing directory.
+         *
+         * @since 2.2.1
+         */
+        public Builder withBasedir(Path basedir) {
+            if (basedir != null) {
+                this.basedir = basedir;
+            } else {
+                this.basedir = DEFAULT_BASEDIR;
```

Comment:
Maybe log a message so it's not silently replaced for someone that didn't read JavaDoc? Same comment for the other new fields.

Reply:
Um, you are right, but this is the ONLY non-null field (all the rest "default with empty collection"). And I'd dislike for override builder to log, hence, am changing this method only: allow non-null and validated (existing directory) parameters only